### PR TITLE
Auto completion for label "name" (Lgraph.open name:)

### PR DESCRIPTION
**Pull request by rabieifk**
check list
* Read the [policy](CONTRIBUTING.md) for the instructions on contributions
* Make sure that the license is BSD-3, MIT, Apache, or similar. Even thought it is not Apache, there should be NO patents in the code
* Add yourself to [credits](CREDITS.txt) if you have a new contribution
* It is recommended to contact a [code owner](CODE_OWNERS.txt) if you are planning a large contribution
* Follow the [github/lgraph](GitHub-use.md) instructions to create a pull request
* Create a meaningful commit log
* In case of conflict, check the [code of conduct](CODE_OF_CONDUCT.md)

---
This change is [Reviewable](https://reviewable.io/reviews/masc-ucsc/livehd/229)

### CODE REVIEW

File: main/main.cpp [hunk 1]
```diff
@@ -19,12 +19,14 @@
 #include "iassert.hpp"
 #include "lgraph.hpp"
 #include "replxx.hxx"
+
+#include "meta_api.hpp"
+#include "../eprp/eprp_var.hpp"
```
Comment (renau) [inline main/main.cpp:24]: Do not do relative path includes. eprp_var.hpp should be in the include path
#include "eprp_var.hpp"

File: main/main.cpp [hunk 2]
```diff
@@ -98,36 +100,58 @@ Replxx::completions_t hook_shared(std::string const& context, int index, std::ve
...
+
+    bool label_files  = strcasecmp(label.c_str(), "files") == 0;
+    bool label_output = strcasecmp(label.c_str(), "output") == 0;
+    bool label_path   = strcasecmp(label.c_str(), "path") == 0;
+    bool label_odir   = strcasecmp(label.c_str(), "odir") == 0;
+    bool label_name   = strcasecmp(label.c_str(), "name") == 0;
+      	
+    if (label_name)
+    {
+    	std::vector<Lgraph *> lgs;
```
Comment (renau) [inline main/main.cpp:124]: Do not open Lgraphs (lgs has no reason to exists there)

File: main/main.cpp [hunk 3]
```diff
@@ -98,36 +100,58 @@ Replxx::completions_t hook_shared(std::string const& context, int index, std::ve
...
+    	std::vector<std::string> name_files;
+    	auto *library = Graph_library::instance("lgdb/");
+
+	    library->each_lgraph([&lgs, &name_files, path](Lg_type_id id, std::string_view name){
+	        (void)name;
+	        name_files.push_back(std::string{name});
+	        Lgraph *lg = Lgraph::open(path, id);
```
Comment (renau) [inline main/main.cpp:131]: Remove this line

File: main/main.cpp [hunk 4]
```diff
@@ -98,36 +100,58 @@ Replxx::completions_t hook_shared(std::string const& context, int index, std::ve
...
+	        lgs.push_back(lg);
```
Comment (renau) [inline main/main.cpp:132]: same

File: main/main.cpp [hunk 5]
```diff
@@ -98,36 +100,58 @@ Replxx::completions_t hook_shared(std::string const& context, int index, std::ve
...
+    bool label_name   = strcasecmp(label.c_str(), "name") == 0;
+      	
+    if (label_name)
+    {
+    	std::vector<Lgraph *> lgs;
+    	std::vector<std::string> name_files;
+    	auto *library = Graph_library::instance("lgdb/");
+
+	    library->each_lgraph([&lgs, &name_files, path](Lg_type_id id, std::string_view name){
+	        (void)name;
```
Comment (renau) [inline main/main.cpp:129]: You are using name, no need to have the '(void)name". this is to indicate the compile that an argument is not used

**Review by renau — CHANGES_REQUESTED** [review]
I wrote some comments. Fix them, and do a new pull requests.

**Comment by rabieifk** [comment]
Sure,
Thank you.

**Pull request merged**